src/v/security/oidc_service.cc

```diff
@@ -72,84 +76,85 @@ struct service::impl {
             co_return;
         }
 
-        namespace coro = ss::coroutine;
-
-        if (auto f = co_await coro::as_future(update_metadata()); f.failed()) {
-            vlog(
-              seclog.error,
-              "Failed to retrieve metadata: {}, error: {}",
-              _discovery_url(),
-              f.get_exception());
-            co_return;
-        }
-
-        if (!_issuer.has_value()) {
-            vlog(seclog.error, "Failed to retrieve issuer from metadata");
-            co_return;
-        }
-        if (!_jwks_url.has_value()) {
-            vlog(seclog.error, "Failed to retrieve jwks_url from metadata");
-            co_return;
-        }
-
-        if (auto f = co_await coro::as_future(update_jwks()); f.failed()) {
-            vlog(
-              seclog.error,
-              "Failed to retrieve jwks: {}, error: {}",
-              _jwks_url.value(),
-              f.get_exception());
-            co_return;
+        auto url_res = security::oidc::parse_url(_discovery_url());
+        if (url_res.has_error()) {
+            throw std::runtime_error("Invalid url");
         }
+        update_url(_parsed_discovery_url, std::move(url_res).assume_value());
 
-        if (!_jwks) {
-            vlog(seclog.error, "Error updating keys: Keys not found");
-            co_return;
-        }
+        co_await update_metadata();
+        co_await update_jwks();
+    }
 
-        auto res = _verifier.update_keys(*_jwks);
-        if (res.has_error()) {
+    ss::future<errc> update_metadata() {
+        auto response_body = co_await ss::coroutine::as_future(
+          make_request(*_parsed_discovery_url));
```

Comment:
> since its assignment and use are not colocated, would it make sense to assert on _parsed_discovery_url containing a value?

I'll colocate them.

> Similar comment for the probe being non-null.

I'm not sure which bit you're referring to.

Reply:
This line (sorry I couldn't figure out how to get a permalink from the core review window): https://github.com/redpanda-data/redpanda/pull/14990/files#r1395970819